### PR TITLE
Generate cases improved

### PR DESCRIPTION
**Pull request by DavertMik**
- Added writing rule for generations skill
- Added prerequisites

### CODE REVIEW

File: skills/generate-cases/references/writing-rule.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+Describe the intent of the test case.
+Intent is needed only if title is not fully enough to explain the intent.
+
+Test case consist of description and steps
+Description must be clear and concise.
+Focus on whitebox testing, thus each operation and observable results must be obtained via public apis or UI.
+Prefer using UI over public apis when possible.
+If you think UI is available, and test can be achived from UI, test must use it
+Understand UI pages and components or API endpoints from source code. If source code not availble ask user for it.
```
Comment (DavertMik) [inline skills/generate-cases/references/writing-rule.md]:
```suggestion
Understand UI pages and components or API endpoints from source code or design or requirements. 
```

File: skills/generate-cases/references/writing-rule.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+## Test Suite Writing Rules
+
+
+Formulas, business rules, edge-case reasoning, and feature
+context live in the suite/test description.
```
Comment (olexandr13) [inline skills/generate-cases/references/writing-rule.md]: ?

File: skills/generate-cases/references/writing-rule.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+Prerequisites which are common to all test cases must be written as bullet points in the suite description.
+Formulas, diagrams, relavant to all test cases can be included as codeblocks in the suite description.
+
+
+## Test Case Writing Rules
+
+Describe the intent of the test case.
+Intent is needed only if title is not fully enough to explain the intent.
```
Comment (olexandr13) [inline skills/generate-cases/references/writing-rule.md]: ?

File: skills/generate-cases/references/writing-rule.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+Test case consist of description and steps
+Description must be clear and concise.
+Focus on whitebox testing, thus each operation and observable results must be obtained via public apis or UI.
```
Comment (olexandr13) [inline skills/generate-cases/references/writing-rule.md]: ?

File: skills/generate-cases/references/writing-rule.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+Prefer using UI over public apis when possible.
+If you think UI is available, and test can be achived from UI, test must use it
+Understand UI pages and components or API endpoints from source code or design or requirements. 
+Add system checks only if it not clear how to test via public apis or UI.
+It is not a unit test, usually no direct server or code access is allowed.
+
+Why in description, what in steps.
```
Comment (olexandr13) [inline skills/generate-cases/references/writing-rule.md]: ?

File: skills/generate-cases/references/writing-rule.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+
+## Steps
+
+Step should consist of **action** (with optional test data) and **expected result**.
+Each step must be simple sentences.
+Avoid using commas, subsentances
+Steps are mechanical: click, send, read, assert.
+Each step must include exact instructions
+Prefer steps that include specific values, urls
```
Comment (olexandr13) [inline skills/generate-cases/references/writing-rule.md]: ?

File: skills/generate-cases/references/writing-rule.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+Prefer concrete values over general words. Replace "small", "known", "around", "e.g.", "like", "(…)" with literal numbers, strings, etc. 
+If no values are availble use placholders variable names like `${url}` or `${company-title}`
+Avoid general statements in steps. Move general statements to the description.
+Do not chain multiple distinct actions or use unnecessary And / Or combinations in a single sequence.
+All step actions must be clear to perform via PUBLIC API or UI
```
Comment (olexandr13) [inline skills/generate-cases/references/writing-rule.md:116]: ?

File: skills/generate-cases/references/writing-rule.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+All checks/verifications/assertions should be in **expected result**, not as separate steps.
```
Comment (olexandr13) [inline skills/generate-cases/references/writing-rule.md]: ?

File: skills/generate-cases/references/writing-rule.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+You may add a codeblock after a step if needed to:
+
+- write API request
+- write SQL
+- write shell command
+- to illustrate the point using pseudocode
+- etc
+
+Avoid using bold/italic formatting in steps.
```
Comment (olexandr13) [inline skills/generate-cases/references/writing-rule.md]: ? why

File: skills/generate-cases/references/writing-rule.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+Puffery: pivotal, crucial, vital, testament, enduring legacy
+Empty "-ing" phrases: ensuring reliability, showcasing features, highlighting capabilities
+Promotional adjectives: groundbreaking, seamless, robust, cutting-edge
+Overused AI vocabulary: delve, leverage, multifaceted, foster, realm, tapestry
+Formatting overuse: excessive bullets, emoji decorations, bold on every other word
+
+Be specific, not grandiose. Say what it actually does. Avoid marketing language.
+
+
+Words to watch: stands/serves as, is a testament/reminder, plays a vital/significant/crucial/pivotal role, underscores/highlights its importance/significance, reflects broader, symbolizing its ongoing/enduring/lasting impact, key turning point, indelible mark, deeply rooted, profound heritage, steadfast dedication...
```
Comment (olexandr13) [inline skills/generate-cases/references/writing-rule.md]: don't understand the purpose of this

**Comment by olexandr13** [comment]
- no BDD (I suggest WWW)
- .md format syntax not fixed

File: skills/generate-cases/references/writing-rule.md (hunk 3 above)
Comment (olexandr13) [reply skills/generate-cases/references/writing-rule.md]: intent should be description

File: skills/generate-cases/references/writing-rule.md (hunk 4 above)
Comment (olexandr13) [reply skills/generate-cases/references/writing-rule.md]: blackbox

Comment (olexandr13) [reply skills/generate-cases/references/writing-rule.md]: unless specified different

Comment (olexandr13) [reply skills/generate-cases/references/writing-rule.md]: prefer FE tests (based on testomat test cases, FE repo)

Comment (olexandr13) [reply skills/generate-cases/references/writing-rule.md]: skill should ask, not suggest

Comment (olexandr13) [reply skills/generate-cases/references/writing-rule.md]: RECOMMEND plan mode

File: skills/generate-cases/references/writing-rule.md (hunk 5 above)
Comment (olexandr13) [reply skills/generate-cases/references/writing-rule.md]: "Why" in description, "what" in steps.

File: skills/generate-cases/references/writing-rule.md (hunk 6 above)
Comment (olexandr13) [reply skills/generate-cases/references/writing-rule.md]: do not specific values unless they are important for scenario (e.g. boundary values)

Comment (olexandr13) [reply skills/generate-cases/references/writing-rule.md]: "url" mean specific **path**, not domain

File: skills/generate-cases/references/writing-rule.md (hunk 8 above)
Comment (olexandr13) [reply skills/generate-cases/references/writing-rule.md]: not in step "action"

File: skills/generate-cases/references/writing-rule.md (hunk 9 above)
Comment (olexandr13) [reply skills/generate-cases/references/writing-rule.md]: rules of applying bold, italic

**Comment by olexandr13** [comment]
less output for user

**Comment by olexandr13** [comment]
think about rendering into html
thus, user interacts with this html and can apply changes, remove some cases

**Review by olexandr13 — APPROVED** [review]
